scripts/plugins/extauth-hook-AD.py

```diff
@@ -41,70 +47,83 @@ def setup_logger():
     # Send to authpriv, which will be redirected to /var/log/secure
     auth_log = logging.handlers.SysLogHandler(
         facility="authpriv", address='/dev/log')
-    logger.addHandler(handler)
-    logger.addHandler(std_err)
-    logger.addHandler(auth_log)
+    log.addHandler(handler)
+    log.addHandler(std_err)
+    log.addHandler(auth_log)
 
 
 setup_logger()
 logger = logging.getLogger(__name__)
 
 
 def run_cmd(cmd, log_cmd=True):
+    """Helper function to run command"""
     try:
         result = subprocess.check_output(cmd)
         if log_cmd:
-            logger.debug("{} -> {}".format(cmd, result))
+            msg = "{} -> {}".format(cmd, result)
+            logger.debug(msg)
         return result.strip()
-    except Exception:
+    except Exception:  # pylint: disable=broad-except
         logger.exception("Failed to run command %s", cmd)
         return None
 
 
 class ADBackend(Enum):
```

Comment:
This does not look like an OO design to me. Should not each backend be modeled as a class that knows the constants it needs to know?

Reply:
> I'm not a Python developer but my main question would be: if we are using classes, why are the different backends not modeled as classes, inheriting from a super class that captures the common behavior? Instead we have branching inside methods based on the backend.

Well, it can be, but I do not find it necessary as it will lead to double number of classes.

Reply:
I'd be inclined to agree with @lindig, this should be a set of three classes

- A common base with everything that is shared
- A legacy PBIS implementation
- A Winbind implementation
otherwise there's a ton of technical debt relating to PBIS and excising it in the future will be much harder than just deleting the class definition

Reply:
Updated, could you please help to take a look. thanks.